app/src/main/java/com/kabouzeid/gramophone/model/lyrics/AbsSynchronizedLyrics.java

```diff
@@ -0,0 +1,55 @@
+package com.kabouzeid.gramophone.model.lyrics;
+
+import android.util.SparseArray;
+
+public abstract class AbsSynchronizedLyrics extends Lyrics {
+    private static final int TIME_OFFSET_MS = 500; // time adjustment to display line before it actually starts
+
+    public final SparseArray<String> lines = new SparseArray<>();
+    public int offset = 0;
+
+    public String getLine(int time) {
+        time += offset + AbsSynchronizedLyrics.TIME_OFFSET_MS;
+
+        int lastLineTime = lines.keyAt(0);
+
+        for (int i = 0; i < lines.size(); i++) {
```

Comment:
`lines` is a `SparseArray` so it does not support foreach loop. It should be iterated using `.keyAt()`/`.valueAt()`